manager.go

```diff
@@ -38,8 +39,17 @@ func (manager *Manager) ProbeAll() {
 
 	manager.LastUpdate = time.Now()
 
+	i := 0.0
+
 	for category, services := range manager.Services {
 		for _, service := range services {
+			// Batching the probes to spread the workload
+			if math.Mod(i, 8) == 0 {
```

Comment:
Question: is there a reason to use a float for `i`?
We could use integers and check with `if i % 8 == 0`.

Moreover, the `i = 0` line might not be necessary (or the condition can just be `i >= 8`).